cpp/src/arrow/flight/sql/odbc/tests/connection_test.cc

```diff
@@ -149,6 +149,162 @@ TEST(SQLFreeConnect, TestSQLFreeConnect) {
+  SQLHENV env;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  // Attempt to set to unsupported version
+  SQLRETURN return_set =
+      SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION, reinterpret_cast<void*>(SQL_OV_ODBC2), 0);
+
+  EXPECT_TRUE(return_set == SQL_SUCCESS);
+}
+
+TEST(SQLSetEnvAttr, TestSQLSetEnvAttrODBCVersionInvalid) {
+  // ODBC Environment
+  SQLHENV env;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  // Attempt to set to unsupported version
+  SQLRETURN return_set =
+      SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION, reinterpret_cast<void*>(1), 0);
+
+  EXPECT_TRUE(return_set == SQL_ERROR);
+}
+
+TEST(SQLGetEnvAttr, TestSQLGetEnvAttrOutputNTS) {
+  // ODBC Environment
+  SQLHENV env;
+
+  SQLINTEGER output_nts;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  SQLRETURN return_get = SQLGetEnvAttr(env, SQL_ATTR_OUTPUT_NTS, &output_nts, 0, 0);
+
+  EXPECT_TRUE(return_get == SQL_SUCCESS);
+
+  EXPECT_EQ(output_nts, SQL_TRUE);
+}
+
+TEST(SQLGetEnvAttr, TestSQLGetEnvAttrGetLength) {
+  // ODBC Environment
+  SQLHENV env;
+
+  SQLINTEGER length;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  SQLRETURN return_get = SQLGetEnvAttr(env, SQL_ATTR_ODBC_VERSION, nullptr, 0, &length);
+
+  EXPECT_TRUE(return_get == SQL_SUCCESS);
+
+  EXPECT_EQ(length, sizeof(SQLINTEGER));
+}
+
+TEST(SQLGetEnvAttr, TestSQLGetEnvAttrNullValuePointer) {
+  // ODBC Environment
+  SQLHENV env;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  SQLRETURN return_get = SQLGetEnvAttr(env, SQL_ATTR_ODBC_VERSION, nullptr, 0, nullptr);
+
+  EXPECT_TRUE(return_get == SQL_ERROR);
+}
+
+TEST(SQLSetEnvAttr, TestSQLSetEnvAttrOutputNTSValid) {
+  // ODBC Environment
+  SQLHENV env;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  // Attempt to set to output nts to supported version
+  SQLRETURN return_set =
+      SQLSetEnvAttr(env, SQL_ATTR_OUTPUT_NTS, reinterpret_cast<void*>(SQL_TRUE), 0);
+
+  EXPECT_TRUE(return_set == SQL_SUCCESS);
+}
+
+TEST(SQLSetEnvAttr, TestSQLSetEnvAttrOutputNTSInvalid) {
+  // ODBC Environment
+  SQLHENV env;
+
+  // Allocate an environment handle
+  SQLRETURN return_env = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(return_env == SQL_SUCCESS);
+
+  // Attempt to set to output nts to unsupported false
+  SQLRETURN return_set =
+      SQLSetEnvAttr(env, SQL_ATTR_OUTPUT_NTS, reinterpret_cast<void*>(SQL_FALSE), 0);
+
+  EXPECT_TRUE(return_set == SQL_ERROR);
+}
+
+TEST(SQLSetEnvAttr, TestSQLSetEnvAttrBadEnv) {
+  // Attempt to set using bad environment pointer
```

Comment:
1. `SQLAllocEnv` is missing 
2. `SQL_OV_ODBC2` is a valid environment attribute

Reply:
We can remove this test